src/sage/schemes/elliptic_curves/ell_point.py

```diff
@@ -778,6 +778,52 @@ def xy(self):
         else:
             return self[0]/self[2], self[1]/self[2]
 
+    def x(self):
+        """
+        Return the `x` coordinate of this point, as an element of the base field.
+        If this is the point at infinity, a :class:`ZeroDivisionError` is raised.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve('389a')
+            sage: P = E([-1,1])
+            sage: P.x()
+            -1
+            sage: Q = E(0); Q
+            (0 : 1 : 0)
+            sage: Q.x()
+            Traceback (most recent call last):
+            ...
+            ZeroDivisionError: rational division by zero
+        """
+        if self[2] == 1:
```

Comment:
I think `a.is_one()` is quite a lot faster than `a == 1`: The reason is that the latter spends time on converting the Python integer `1` into the same base ring as `a`.

Reply:
Fixed. I also modified the method `.xy()` both to keep them equal and for the same speed reason